variant_remapping_tools/reads_to_remapped_variants.py

```diff
@@ -64,32 +63,55 @@ def is_read_valid(read, counter, flank_length, score_cutoff, diff_cutoff):
     return True
 
 
-def process_bam_file(bam_file_path, output_file, old_ref_allele_output_file, flank_length, score_perc, diff_AS_XS_perc):
+def fetch_bases(fasta, contig, length, start):
+    zero_base_start = start - 1
+    end = zero_base_start + length
+    new_ref = pysam.FastaFile(fasta).fetch(reference=contig, start=zero_base_start, end=end)
+    return new_ref
+
+
+def calculate_new_alleles(old_ref, new_ref, old_alt, is_reverse_strand):
+    # If reverse strand -> calculate complement
+    old_ref_processed = old_ref
+    old_alt_processed = old_alt
+    if is_reverse_strand:
+        old_ref_processed = Seq(old_ref_processed, generic_dna).complement()
+        old_alt_processed = Seq(old_alt_processed, generic_dna).complement()
+
+    # No changes
+    if old_ref_processed == new_ref:
+        return new_ref, old_alt_processed
+    # Ref and Alt are the same -> Alt changed to old Ref
+    if new_ref == old_alt_processed:
+        return new_ref, old_ref_processed
+    else:
+        return new_ref, old_alt_processed
+
+
+def process_bam_file(bam_file_path, output_file, flank_length, score_perc, diff_AS_XS_perc, new_genome):
     bamfile = pysam.AlignmentFile(bam_file_path, 'rb')
 
     # Calculate the score cutoff based on flanking seq length
     score_cutoff = -(flank_length * score_perc)
     diff_cutoff = flank_length * diff_AS_XS_perc
     counter = Counter()
 
-    # Reverses the allele for variants that got mapped onto the reverse strand of the new genome, and prints everything
-    # into correct columns
-    with open(output_file, 'w') as outfile, open(old_ref_allele_output_file, 'w') as old_ref_alleles:
+    with open(output_file, 'w') as outfile:
         for read in bamfile:
-            if is_read_valid(read, counter, flank_length, score_cutoff, diff_cutoff):
+            varpos, perf_counter, local_region_size = calculate_variant_position(read, flank_length)
+            if is_read_valid(read, counter, flank_length, score_cutoff, diff_cutoff, perf_counter, local_region_size):
```

Comment:
Calculating the variant position before the filtering will fail when encountering unmapped sequence.
```suggestion
            if is_read_valid(read, counter, flank_length, score_cutoff, diff_cutoff, perf_counter, local_region_size):
                varpos, perf_counter, local_region_size = calculate_variant_position(read, flank_length)
```